src/main/java/com/google/gcp_variant_transforms/library/VariantToBqUtilsImpl.java

```diff
@@ -199,29 +196,30 @@ public void addFormatAndPhaseSet(TableRow row, Genotype genotype, VCFHeader vcfH
         continue; // We will set GT field in a separate "genotype" field in BQ row.
       }
       if (genotype.hasAnyAttribute(attrName)) {
+        String sanitizedFieldName = SchemaUtils.getSanitizedFieldName(attrName);
         Object value = genotype.getAnyAttribute(attrName);
-        VCFFormatHeaderLine formatMetadata = vcfHeader.getFormatHeaderLine(attrName);
-        VCFHeaderLineType formatType = formatMetadata.getType();
+        VCFHeaderLineType formatType = formatHeaderLine.getType();
         if (attrName.equals(VCFConstants.GENOTYPE_ALLELE_DEPTHS) ||
             attrName.equals(VCFConstants.DEPTH_KEY) ||
             attrName.equals(VCFConstants.GENOTYPE_QUALITY_KEY) ||
             attrName.equals(VCFConstants.GENOTYPE_PL_KEY)) {
           // These four field values have been pre-processed in Genotype.
-          row.set(attrName, value);
+          row.set(sanitizedFieldName, value);
         } else if (attrName.equals(VCFConstants.PHASE_SET_KEY)) {
           String phaseSetValue = genotype.getAnyAttribute(VCFConstants.PHASE_SET_KEY).toString();
           phaseSet = replaceMissingWithNull(phaseSetValue);
         } else {
           // The rest of fields need to be converted to the right type as VCFHeader specifies.
-          VCFHeaderLineCount formatCountType = formatMetadata.getCountType();
+          VCFHeaderLineCount formatCountType = formatHeaderLine.getCountType();
           if (formatCountType == VCFHeaderLineCount.INTEGER) {
-            row.set(attrName, convertToDefinedType(genotype.getAnyAttribute(attrName),
-                formatType, formatMetadata.getCount()));
+            row.set(sanitizedFieldName, convertToDefinedType(genotype.getAnyAttribute(attrName),
+                formatType, formatHeaderLine.getCount()));
           } else {
```

Comment:
This else statement covers all fields that `formatCountType != VCFHeaderLineCount.INTEGER` which means 'A', 'R', 'G', and '.'.
* Let's update the comment.
* Why we are not checking the count the same way we did for INFO fields?

Reply:
comments updated. For INFO field, we need to deal with A and R since we have an expected count and we need to split field number = A to `alt` field. As we have discussed before, in `FORMAT`, if we met those count , we handle as number = ".", and do not check the count of the field.